### PR TITLE
[proxy] Add support for Diffie–Hellman key exchange

### PR DESCRIPTION
**Pull request by aledbf**
How to test:

- ensure the werft build is green
- ensure the in-workspace build works
- open browser https://aledbf-ssl-dhparam.staging.gitpod-dev.com/workspaces/

Currently, DHE suites not supported - https://www.ssllabs.com/ssltest/analyze.html?d=gitpod.io&hideResults=on

### CODE REVIEW

File: .werft/values.dev.yaml [hunk 1]
```diff
@@ -103,6 +103,7 @@ components:
     serviceType: "ClusterIP"
     deployIngressService: false
     loadBalancerIP: null
+    sslDHParam: LS0tLS1CRUdJTiBESCBQQVJBTUVURVJTLS0tLS0KTUlJQkNBS0NBUUVBdkhsWGFIOHlyNWFJYWhmc3N5RDltTDJXL2NQZUdhWmtwdUF6dm1KamFXeEZqUFJrM2IvRworbDNJY3pTRndVVy9YN2NjU2NlZ1JzZjl1dFJKR1hWZWMyL0c1VFlETkJsNDdMQ1I1VXMvUmpsV0tHcWJTVnVhCnYxY3BsUUgvVDl4UGJ4bjE2QjhrOFZmdEVUNEgxTmhRM2lxSWtGbUk2QTZrTDJzbi9VQjdhMXYydDRjL3VpdVcKcGtNNlVaRlVoUGhhVkt6VWJsR0pTWEdnbjdiVUtGWDNQSVdLY1VHM015NmhYVWg0TzRLTXhaYWRWS051T011SApyK0JLZDdsWkx5SmlBQWRtWUtTaEpNVGUwSS9uRTlFR3Q0VHVGZVZNaXpkbHR5SWkyWEFLRUlwYjZuTFlHWjY1CkNPRkx2WFpXOTJOK0U0ZFdES1ozaHBTbk1qclptVEIvYXdJQkFnPT0KLS0tLS1FTkQgREggUEFSQU1FVEVSUy0tLS0tCg==
```
Comment (csweichel) [inline .werft/values.dev.yaml:106]: I'm not sure I understand the role of this parameter properly, but wonder if couldn't fold this into the certificate secret? AFAIK we even have a `params` file there.

Comment (aledbf) [reply .werft/values.dev.yaml:106]: @csweichel a Diffie-Hellman parameter is a set of randomly generated data used when establishing Perfect Forward Secrecy during initiation of an HTTPS connection. Is not related to the SSL certificate.
Without this setting, DHE ciphers cannot be used. http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_dhparam

This post provides more details https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html#toc_5

Comment (geropl) [reply .werft/values.dev.yaml:106]: >  Is not related to the SSL certificate.

Absoluteley, makes sense.

File: chart/values.yaml [hunk 2]
```diff
@@ -367,6 +368,10 @@ components:
     certbot:
       enabled: false
       email: "certificate@your-domain.com"
+    # A base64ed Diffie-Hellman parameter
+    # This can be generated with: openssl dhparam 4096 2> /dev/null | base64 -w 0
+    # http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_dhparam
```
Comment (geropl) [inline chart/values.yaml:373]: :100:  :+1:

Comment (geropl) [reply chart/values.yaml:373]: With this IMO we do not need to update the docs (also a nice side effect of the DH/cert separation).

**Review by geropl — APPROVED** [review]
Cool, finally DH params again! :tada:

**Pull request merged**